grails-app/controllers/org/pih/warehouse/order/OrderController.groovy

```diff
@@ -656,16 +656,22 @@ class OrderController {
             render(status: 500, text: "${warehouse.message(code: 'orderItem.blockedSupplier.label')}")
             return
         }
+        if (params.productSupplier || params.supplierCode) {
+            productSupplier = productSupplierDataService.getOrCreateNew(params)
+        }
         if (params.productSupplier == "Create New") {
             Organization supplier = Organization.get(params.supplier.id)
             productSupplier = ProductSupplier.findByCodeAndSupplier(params.sourceCode, supplier)
             if (params.sourceCode && productSupplier) {
                 render(status: 500, text: "Product source with given code for your supplier already exists")
                 return
             }
-        }
-        if (params.productSupplier || params.supplierCode) {
-            productSupplier = productSupplierDataService.getOrCreateNew(params)
+            try {
+                productSupplier = productSupplierDataService.createProductSupplierWithoutPackage(params)
```

Comment:
You are basically creating the productSupplier for the second time here.

Reply:
In this case you can create without having enter any data for new product supplier or without having product supplier chosen before. In the previous if you can only get already available product supplier or create new with given data.

Reply:
1. `productSupplierDataService.getOrCreateNew(params)` will call `productSupplierDataService.createProductSupplierWithoutPackage(params)` inside. So it's triggered twice.
2. If you will have params.productSupplier == "Create New" and params.sourceCode, then the line 666 will trigger. (first, you create it in line 660, and then line 664 will find it and will complain that it already exists).
3. Indentation of try-catch seems to be a little off.